dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/events/importer/context/CategoryOptionComboSupplier.java

```diff
@@ -86,8 +95,13 @@ public CategoryOptionComboSupplier( NamedParameterJdbcTemplate jdbcTemplate, Pro
             }
 
             final CategoryCombo programCatCombo = program.getCategoryCombo();
-            final String aoc = event.getAttributeOptionCombo();
-            final String attributeCatOptions = event.getAttributeCategoryOptions();
+            String aoc = event.getAttributeOptionCombo();
+            String attributeCatOptions = event.getAttributeCategoryOptions();
+
+            if ( psi != null )
+            {
+                aoc = programStageInstanceMap.get( event.getUid() ).getAttributeOptionCombo().getUid();
```

Comment:
What is the payload we are getting? If we are getting aoc related info, we are completely ignoring it and using what is already stored.

Will it be better if we could compare incoming with existing and reject if they do not match? That way we give user accurate feedback?

Reply:
we are not getting aoc in payload while we update event. This was the reason for failing. When we update then we should get aoc information from the PSI.